sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateStoreCheckpointFormatV2Suite.scala

```diff
@@ -211,6 +221,323 @@ class RocksDBStateStoreCheckpointFormatV2Suite extends StreamTest
     }
   }
 
+  val changelogEnabled =
+    "spark.sql.streaming.stateStore.rocksdb.changelogCheckpointing.enabled" -> "true"
+  val changelogDisabled =
+    "spark.sql.streaming.stateStore.rocksdb.changelogCheckpointing.enabled" -> "false"
+  val ckptv1 = SQLConf.STATE_STORE_CHECKPOINT_FORMAT_VERSION.key -> "1"
+  val ckptv2 = SQLConf.STATE_STORE_CHECKPOINT_FORMAT_VERSION.key -> "2"
+
+  val testConfigSetups = Seq(
+    // Enable and disable changelog under ckpt v2
+    (Seq(changelogEnabled, ckptv2), Seq(changelogEnabled, ckptv2)),
+    (Seq(changelogDisabled, ckptv2), Seq(changelogDisabled, ckptv2))
+  )
```

Comment:
can you please create a case class for these:

```scala
case class ChangelogCompatibilityTest(
  configsForFirstRun: Seq[(String, String)],
  configsForSecondRun: Seq[(String, String)]
) {
  override def toString: String = first run: (changeLogEnabled, ckpt ver): " +
      ...
}
```

then use the `toString` for the test names